flink/v2.0/flink/src/main/java/org/apache/iceberg/flink/maintenance/operator/DataFileRewriteCommitter.java

```diff
@@ -128,6 +142,14 @@ public void processWatermark(Watermark mark) throws Exception {
     try {
       if (commitService != null) {
         commitService.close();
+
+        if (collectResults) {
+          RewriteDataFilesActionResult actionRes =
+              new RewriteDataFilesActionResult(
+                  Lists.newArrayList(removedDataFiles.iterator()),
+                  Lists.newArrayList(addedDataFiles.iterator()));
+          output.collect(new StreamRecord<>(Trigger.create(actionRes)));
+        }
```

Comment:
It's kind of odd we only generate results in batch mode. Would it make sense to always generate them? This would allow to tap into the results as part of a streaming job, or when we verify in test results.